### PR TITLE
Add timeout to gm call.

### PR DESCRIPTION
**Pull request by mgaeta**
Pass the `timeout` option to gm and consume the resulting error. Also test the desired behavior.

### CODE REVIEW

File: lib/pdf2img.js [hunk 1]
```diff
@@ -47,8 +52,12 @@ Pdf2Img.prototype.convert = function(input, opts, callbackreturn) {
     opts = {};
   }
   var options = getOptions(opts);
+
+  // Set new timeout.
+  gm.prototype.options({timeout: options.timeout});
```
Comment (ericyhwang) [inline lib/pdf2img.js]: Let's not set options onto the prototype, given that I just did some work to remove global state. Does chaining it not work? E.g. `gm(input).options({timeout: timeout}).whatever(...)`?

Comment (mgaeta) [reply lib/pdf2img.js]: good point, i'll try it.

**Comment by mgaeta** [comment]
Ok, I've set the option correctly and also cleaned up the commits.

File: lib/pdf2img.js [hunk 2]
```diff
@@ -79,43 +85,52 @@ Pdf2Img.prototype.convert = function(input, opts, callbackreturn) {
...
-
+    function (callback) {
+      gm(input)
+        .options({timeout: options.timeoutMilliseconds})
+        .identify("%p ", function (err, value) {
+          if (err) {
+            if (err.message === 'gm() resulted in a timeout.') {
+              return callback({
+                result: 'timeout',
+                message: 'gm call took longer than ' + options.timeoutMilliseconds + 'ms.'
```
Comment (ericyhwang) [inline lib/pdf2img.js]: Would be good to specify `gm identify` here and `gm convert` below, to indicate which step timed out.

File: test/index.js [hunk 3]
```diff
@@ -74,19 +86,32 @@ describe('Split and convert pdf into images', function() {
   });
   it ('The timeout parameter will end conversion', function(done) {
     this.timeout(100000);
+    var milliseconds = 10;
+    var opts = {timeoutMilliseconds: milliseconds};
+    var error = {
+      result: 'timeout',
+      message: 'gm identify call took longer than '+ milliseconds +'ms.'
```
Comment (ericyhwang) [inline test/index.js]: Minor linting thing, spaces around plus signs.

**Review by ericyhwang — APPROVED** [review]

**Pull request merged**